server.cc

```diff
@@ -104,13 +113,16 @@ static bool is_valid_method(std::string method)
 }
 
 // return a random err_no
-static int random_err_no()
+static int random_err_no(std::vector<int32_t> err_nos)
 {
-    std::random_device rd;
-    std::uniform_int_distribution<int> dist(E2BIG, EXFULL);
+    if (err_nos.empty() == 0) {
+        err_nos = default_errnos;
+    }
 
-    return dist(rd);
-} 
+    std::random_device rd;
+    std::uniform_int_distribution<int> dist(0, err_nos.size());
+    return int(err_nos[dist(rd)]);
```

Comment:
As a follow up, we need to use a PRNG (`default_random_engine`) here rather than std::random_device.
The std::random_device that is often hardware based is too expensive and slow if used too frequently once its entropy pool is exhausted.  It should be used just to get a seed for initializing the default_random_engine.

It would also be nice to add an option to provide a seed to have reproducible runs.